src/lib/api/projects.ts

```diff
@@ -151,15 +340,23 @@ export async function documents(
   endpoint.searchParams.set("ordering", "-created_at");
   endpoint.searchParams.set("per_page", "12");
 
-  const res = await fetch(endpoint, { credentials: "include" }).catch((e) => {
-    error(500, { message: e });
-  });
+  const resp = await fetch(endpoint, { credentials: "include" }).catch(
+    console.error,
+  );
+
+  if (!resp) {
+    throw new Error("API error");
+  }
 
-  if (isErrorCode(res.status)) {
-    error(res.status, {
-      message: res.statusText,
-    });
+  if (isErrorCode(resp.status)) {
+    throw new Error(resp.statusText);
   }
 
-  return res.json();
+  return resp.json();
+}
+
+// utils
+
+export function canonicalUrl(project: Project): URL {
```

Comment:
This this is frontend-related logic and doesn't have much to do with the Projects API, I think this belongs somewhere else…not sure where, though. Maybe a `utils/urls.ts` file?

Reply:
This is how we've organized functions like this for documents and notes, so I think this is the right place for it.